### ISSUE TITLE
gRPC number of concurrent stream block at 250

### ISSUE THREAD

**Issue by MaxThom** [traefik/traefik#8768]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

I created a simple gRPC client and server. The server is deployed inside our k8s cluster (with k3s) and traefik v2.4.8.
There is only one replica of the server. I got an ingress using h2c protocol and everything works great.

This is my proto, a simple stream from the server to the client:
```proto
service TelemetryService {
  rpc GetTelemetry(TelemetryMeta) returns (stream Telemetry) {}
}
```

When I create a large number of stream from the client by calling GetTelemetry(), the number get blocked to exactly 250.
If I use a NodePort instead of the ingress, everything works as expected.

Moreover, when I close the client, I dont receive the disconnect message in the server. Again, if I use the NodePort, it works.

Some application we have can use up to 5000 concurrent steam.

### What did you see instead?

1) Limit of 250 of concurrent stream
2) No disconnect message in the server when the client is closed

### What version of Traefik are you using?

traefik v2.4.8 with helm chart traefik-9.18.2

### What is your environment & configuration?

```yaml
ports:                                                                                                                                                                                                            
  grpc:                                                                                                                                                                                                  
    port: 50051                                                                                                                                                                                                    
    protocol: TCP                                                                                                                                                                                                  
    expose: true                                                                                                                                                                                                   
    exposedPort: 50051
```

The rest is default from the helm chart.

We use kubernetes with k3s 1.21 deployed on ubuntu 20.04.


### If applicable, please paste the log output in DEBUG level

_No response_

**Comment by tomMoulard** [comment]
Hello @MaxThom,

Thanks for your interest in Traefik!

Indeed, after further investigating, there is a limit to concurrent streams set by Go.

The PR #8781 allows changing the default value.
This would allow more than 250 concurrent streams.

**Comment by MaxThom** [comment]
Cool thank you Tom! Will that config be easy to set in Kubernetes with a Middleware or ServiceTransport? Or in the entrypoint in the values file of the chart? Looking foward to this and deploy it to all our clusters!

Thanks a lot !

**Comment by tomMoulard** [comment]
This fix will be accessible in a future bug fix release (most probably `v2.6.2`).

As for Kubernetes, until the [Traefik helm chart](https://github.com/traefik/traefik-helm-chart) is updated, the way to do it will be with `additionalArguments` (e.g. https://github.com/traefik/traefik-helm-chart/blob/cc2a4bbbc58906cf51a3ef2cff7493185cc5ec1a/traefik/values.yaml#L224-L230, `helm install --set="additionalArguments={--entrypoints.web.http2.maxconcurrentstreams=5000}"`).

This option could be accessible under the entrypoint configuration in the value file if a PR handles it.

**Comment by chuckyz** [comment]
hey @tomMoulard I believe I'm hitting this as well.  Have you done testing on #8781 that the settings frame is being properly propagated throughout the entrypoint/router/service as expected?  We're doing some testing now and finding difficulty understanding where some of these numbers are coming from.

Also, is it clear what Traefik is using for max concurrent streams when it's opening connections internally _to_ services?  Can we possibly override that as well?  I've been looking through net/http and it seems it's almost always set to 100 unless PUSH_PROMISE is set to 1, which I don't know of any circumstances where that _will_ be set to 1.

**Comment by MaxThom** [comment]
Hey @tomMoulard, I'm really happy for this new feature. I see the PR was merged. Does this mean it will be available in v2.6.4?
or v2.7 or further like in the PR comment?
Thanks for all the work.

**Comment by tomMoulard** [comment]
As this PR has been merged on `master`, it will be released in the next minor version : `v2.8`.
